modules/gapi/include/opencv2/gapi/infer/ie.hpp

```diff
@@ -25,6 +25,22 @@ namespace ie {
 
 GAPI_EXPORTS cv::gapi::GBackend backend();
 
+/**
+ * Specify how G-API and IE should trait input data
+ *
+ * In OpenCV, the same cv::Mat is used to represent both
+ * image and tensor data. Sometimes those are hardly distinguishable,
+ * so this extra parameter is used to give G-API a hint.
+ *
+ * This hint controls how G-API reinterprets the data when converting
+ * it to IE Blob format (and which layout/etc is assigned to this data).
+ */
+enum class TraitAs: int
+{
+    TENSOR, //!< G-API traits an associated cv::Mat as a raw tensor and passes dimensions as-is
+    IMAGE   //!< G-API traits an associated cv::Mat as an image so creates an "image" blob (NCHW/NHWC, etc)
```

Comment:
Alternative is using separate methods:
- `constImageInput()`
- `constTensorInput()`

Reply:
Good idea, will consider, thanks!